tests/dataset_runner.py

```diff
@@ -148,27 +177,120 @@ def complete_submission(self):
         if response.status_code != requests.codes.accepted:
             raise RuntimeError(f"PUT {submit_url} returned {response.status_code}: {response.content}")
         Progress.report("  done.\n")
-
-    def wait_for_primary_and_results_bundles(self):
+    
+    def wait_for_primary_bundles_analysis_workflows_and_results_bundles(self):
         """
-        We used to wait for all primary bundles to be created before starting to look for results.
+        We wait for all primary bundles to be created before starting to look for results in non-scaling tests.
         It appears that with large submissions, results can start to appear before bundle export is finished,
-        so we now monitor both kinds of bundles simultaneously.
+        so we monitor both kinds of bundles simultaneously in scaling tests.
         """
-        Progress.report("WAITING FOR PRIMARY AND RESULTS BUNDLE(s) TO BE CREATED...")
+        Progress.report("WAITING FOR PRIMARY BUNDLE(s), ANALYSIS WORKFLOWS AND RESULTS BUNDLE(s)...")
         self.expected_bundle_count = self.dataset.config["expected_bundle_count"]
         WaitFor(
-            self._count_primary_and_secondary_bundles
+            self._count_primary_bundles_analysis_workflows_and_results_bundles_and_report
```

Comment:
Completely optional: you could get rid of the `and_report` to make the function name a bit shorter

Reply:
done!